cmake/fake_tribits.cmake

```
@@ -0,0 +1,331 @@
+ELSEIF(TARGET ${TARGET}) 
+  #the target actually exists - this means we are doing separate libs
+  #or this a test library
+  KOKKOS_LIB_TYPE(${TARGET} INCTYPE) 
+  TARGET_INCLUDE_DIRECTORIES(${TARGET} ${INCTYPE} ${ARGN})
+ELSE()
+  GET_PROPERTY(LIBS GLOBAL PROPERTY KOKKOS_LIBRARIES_NAMES)
+  IF (${TARGET} IN_LIST LIBS)
+     SET_PROPERTY(GLOBAL APPEND PROPERTY KOKKOS_LIBRARY_INCLUDES ${ARGN})
+  ELSE()
+    MESSAGE(FATAL_ERROR "Trying to set include directories on unknown target ${TARGET}")
+  ENDIF()
+ENDIF()
+ENDFUNCTION(KOKKOS_TARGET_INCLUDE_DIRECTORIES TARGET)
+
+FUNCTION(KOKKOS_LINK_INTERNAL_LIBRARY TARGET DEPLIB)
+IF(KOKKOS_HAS_TRILINOS)
+  #do nothing
+ELSE(KOKKOS_SEPARATE_LIBS)
+  SET(options INTERFACE)
+  SET(oneValueArgs)
+  SET(multiValueArgs)
+  CMAKE_PARSE_ARGUMENTS(PARSE 
+    "INTERFACE"
+    ""
+    ""
+    ${ARGN})
+  SET(LINK_TYPE)
+  IF(PARSE_INTERFACE)
+    SET(LINK_TYPE INTERFACE)
+  ELSE()
+    SET(LINK_TYPE PUBLIC)
+  ENDIF()
+  TARGET_LINK_LIBRARIES(${TARGET} ${LINK_TYPE} ${DEPLIB})
+  VERIFY_EMPTY(KOKKOS_LINK_INTERNAL_LIBRARY ${PARSE_UNPARSED_ARGUMENTS})
+ENDIF()
+ENDFUNCTION(KOKKOS_LINK_INTERNAL_LIBRARY)
+
+FUNCTION(KOKKOS_ADD_TEST_LIBRARY NAME)
+IF (KOKKOS_HAS_TRILINOS)
+  TRIBITS_ADD_LIBRARY(${NAME} ${ARGN} TESTONLY
+   ADDED_LIB_TARGET_NAME_OUT ${NAME}
+  )
+ELSE()
+  SET(oneValueArgs)
+  SET(multiValueArgs HEADERS SOURCES)
+
+  CMAKE_PARSE_ARGUMENTS(PARSE 
+    "STATIC;SHARED"
+    ""
+    "HEADERS;SOURCES"
+    ${ARGN})
+
+  IF(PARSE_HEADERS)
+    LIST(REMOVE_DUPLICATES PARSE_HEADERS)
+  ENDIF()
+  IF(PARSE_SOURCES)
+    LIST(REMOVE_DUPLICATES PARSE_SOURCES)
+  ENDIF()
+  ADD_LIBRARY(${NAME} ${PARSE_SOURCES})
+  target_compile_options(
+    ${NAME}
+    PUBLIC $<$<COMPILE_LANGUAGE:CXX>:${KOKKOS_CXX_FLAGS}>
+  )
+  target_link_libraries(
+    ${NAME}
+    PUBLIC ${KOKKOS_LD_FLAGS}
+  )
+ENDIF()
+ENDFUNCTION(KOKKOS_ADD_TEST_LIBRARY)
+
+
+FUNCTION(KOKKOS_TARGET_COMPILE_DEFINITIONS)
+  IF (KOKKOS_HAS_TRILINOS)
+    TARGET_COMPILE_DEFINITIONS(${TARGET} ${ARGN})
+  ELSE()
+    TARGET_COMPILE_DEFINITIONS(${TARGET} ${ARGN})
+  ENDIF()
+ENDFUNCTION(KOKKOS_TARGET_COMPILE_DEFINITIONS)
+
+FUNCTION(KOKKOS_INCLUDE_DIRECTORIES)
+IF(KOKKOS_HAS_TRILINOS)
+  TRIBITS_INCLUDE_DIRECTORIES(${ARGN})
+ELSE()
+  CMAKE_PARSE_ARGUMENTS(
+    INC
+    "REQUIRED_DURING_INSTALLATION_TESTING"
+    ""
+    ""
+    ${ARGN}
+  )
+  INCLUDE_DIRECTORIES(${INC_UNPARSED_ARGUMENTS})
+ENDIF()
+ENDFUNCTION(KOKKOS_INCLUDE_DIRECTORIES)
+
+
+MACRO(KOKKOS_ADD_COMPILE_OPTIONS)
+ADD_COMPILE_OPTIONS(${ARGN})
+ENDMACRO()
+
+MACRO(PRINTALL match)
+get_cmake_property(_variableNames VARIABLES)
+list (SORT _variableNames)
+foreach (_variableName ${_variableNames})
+  if("${_variableName}" MATCHES "${match}")
+    message(STATUS "${_variableName}=${${_variableName}}")
+  endif()
+endforeach()
+ENDMACRO(PRINTALL)
+
+MACRO(GLOBAL_APPEND VARNAME)
+  SET(TEMP ${VARNAME})
+  LIST(APPEND TEMP ${ARGN})
+  GLOBAL_SET(${VARNAME} ${TEMP})
```

```suggestion
  SET(VARNAME "${VARNAME};${TEMP}" CACHE INTERNAL "")
```
This didn't work for me. I ended up with `KOKKOS_CUDA_OPTIONS` (literally) showing up in the compile arguments.